ogc-rs/src/wpad.rs

```diff
@@ -0,0 +1,129 @@
+//! The ``wpad`` module of ``ogc-rs``.
+//!
+//! This module implements a safe wrapper around the wii controller functions found in ``wpad.h``.
+
+use alloc::boxed::Box;
+use core::mem;
+
+/// Represents the wpad service. No wii remotes can be read until an instance of
+/// this struct is created. This service can only be created once!
+pub struct Wpad;
+
+/// The remote to be read for the `wpad` service.
+#[derive(Copy, Clone)]
+pub enum Remote {
+    One = 0,
+    Two = 1,
+    Three = 2,
+    Four = 3,
+}
```

Comment:
`Remote` and `Controller` for `wpad` and `pad` are different but represent a common thing. I think you should choose one over the other.

Reply:
I'd use Controller; it's a more generic one (GCN controller isn't a remote)